### PR TITLE
fix: cancel in progress CI runs

### PR DESCRIPTION
**Pull request by vinitkumar**
This improves the CI experience as it will run the github checks in a
group and at the same time cancel in progress jobs if a new commit is
pushed for the pull request.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow

No only does it makes the contributing experience nice, but it also
saves tonnes of energy.

Also, update action-python and action-checkout and cache pip.

- Github Issue: #8276

Authored-by: Vinit Kumar <vinit.kumar@kidskonnect.nl>
Signed-off-by: Vinit Kumar <vinit.kumar@kidskonnect.nl>

_Please check the following:_

- [x] Do the tests still pass?[^1]
- [x] Does the code comply with the style guide? 
    - [ ] Run `make lint` from the Wagtail root. 
- [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
- [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    - [ ] **Please list the exact browser and operating system versions you tested**:
    - [ ] **Please list which assistive technologies [^3] you tested**: 
- [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**. 

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by lb-** [comment]
@vinitkumar thanks so much for submitting this PR - love it!

I'm not an expert at GitHub actions so I'll let another core team member review but I have approved this CI running on this PR so you can check it all works.

**Comment by vinitkumar** [comment]
> @vinitkumar thanks so much for submitting this PR - love it!
> 
> I'm not an expert at GitHub actions so I'll let another core team member review but I have approved this CI running on this PR so you can check it all works.

Thanks for taking the time. I am happy to answer any questions about them. We have the same optimization running on Django CMS Github actions, so I am pretty confident they will just work.

**Comment by zerolab** [comment]
Hey both,

The changes by and large look good to me. Anything for speeding up CI and saving processing time!

The `cache: 'pip'` bit likely needs more work in that we do not have an explicit [`requirements.txt` file](https://github.com/actions/setup-python#caching-packages-dependencies) -- all dependencies are in `setup.py`

We do a cache based on `setup.py` in the [Circle CI configuration](https://github.com/wagtail/wagtail/blob/main/.circleci/config.yml#L19)

**Comment by vinitkumar** [comment]
> Hey both,
> 
> The changes by and large look good to me. Anything for speeding up CI and saving processing time. Thank you @vinitkumar!
> 
> The `cache: 'pip'` bit likely needs more work in that we do not have an explicit [`requirements.txt` file](https://github.com/actions/setup-python#caching-packages-dependencies) -- all dependencies are in `setup.py`
> 
> We do a cache based on `setup.py` in the [Circle CI configuration](https://github.com/wagtail/wagtail/blob/main/.circleci/config.yml?rgh-link-date=2022-04-05T08%3A09%3A05Z#L19)

Thanks!

Do you want me to revert the 'cache: 'pip'` related change?

**Comment by zerolab** [comment]
> Do you want me to revert the 'cache: 'pip'` related change?

Let's do that and revisit in a separate PR

**Comment by vinitkumar** [comment]
@zerolab Updated the PR description and reverted the change for pip caching. Should be good now. Let me know if you need any more changes.

**Review by zerolab — APPROVED** [review]
LGTM!

**Pull request merged**
